tests/bluetooth/controller/ctrl_unsupported/testcase.yaml

```diff
@@ -3,3 +3,15 @@ common:
 tests:
   bluetooth.controller.ctrl_unsupported.default.test:
      type: unit
+  bluetooth.controller.ctrl_unsupported.no_enc.test:
+     type: unit
+     extra_args: NO_ENC=y KCONFIG_OVERRIDE_FILE="kconfig_override.h"
```

Comment:
seems like I was not invited to this party #40073

The way `KCONFIG_OVERRIDE_FILE` and `tests/bluetooth/controller/common/defaults_cmake.txt` appears to be short cutting everything else we do in the build system.

But guess that is outside the scope of this PR.